app/controllers/auth_controller.rb

```diff
@@ -26,10 +29,20 @@ def token
         token_type: 'Bearer',
         expires_in: time_to_live,
         scope: scope,
-        patient: Patient.all.first.id
+        patient: Patient.first&.id
       }
+    elsif params.key? :code
```

Comment:
We also need to check `:client_id` in `def token`. `code` and `client_id` are two separate params.

Reply:
I changed it so it will be:
```ruby
elsif params.key? :code or params.key? :client_id
      (return 400 bad_request with error: invalid_client)
else
      (return 400 bad_request with error: invalid_request)
```
but is this correct? To my understanding client_id should only be passed to /auth/authorize, and if its sent to /auth/token the error message should be error: invalid_request

Reply:
(those return statements are pseudo-code btw)

Reply:
Like Steve said below since this is the [public app workflow](http://hl7.org/fhir/smart-app-launch/#step-3-app-exchanges-authorization-code-for-access-token), both `code` and `client_id` are required. So we should check if `client_id` is equal to `Rails.application.config.client_id`, same check in `/auth/authorize`

Reply:
got it